spec/lcm/integration/user_provisioning_end_to_end_spec.rb

```diff
@@ -6,56 +6,81 @@
 require_relative 'support/s3_helper'
 require_relative 'brick_runner'
 
-def uniq_label_with_value(project)
+def uniq_label_with_value(project, value_count)
   blacklists = $generated_uniq_labels_with_value || {}
   blacklist = blacklists[project.pid] || []
 
-  atts = project.attributes.to_a
+  atts = project.attributes.to_a.sort_by(&:identifier)
   attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || (blacklist.include? attribute)
   label = attribute.primary_label
-  value = label.values.find { |v| !v[:value].empty? }
+  sorted_values = label.values.to_a.sort { |x, y| x[:value] <=> y[:value] }
+  notempty = sorted_values.select { |v| !v[:value].empty? }
+  values = notempty.slice(0, value_count)
 
   blacklists[project.pid] = blacklist.push attribute
   $generated_uniq_labels_with_value = blacklists
-  [label, value[:value]]
+  [label, values.map { |v| v[:value] }]
 end
 
-shared_context 'user filters brick' do
-  it 'updates user filters' do
-    BrickRunner.user_filters_brick context: @test_context, template_path: '../params/user_filters_brick_e2e.json.erb'
+def set_up_random_muf_data(project, user_data, complexity = 2)
+  label, values = uniq_label_with_value project, complexity
+  puts "Settings up MUF referencing value #{values} of attribute #{label.title}"
+  mufs = user_data.map do |u|
+    values.map do |v|
+      {
+        login: u[:custom_login],
+        client_id: u[:client_id],
+        value: v,
+        label_id: label.identifier
+      }
+    end
+  end.flatten
+  Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(mufs), Support::S3Helper::USER_FILTERS_KEY)
+  mufs
+end
 
-    @projects.each do |p|
-      filters = p.user_filters.to_a
-      expect(filters.length).to eq(@mufs.count / @projects.count)
-      @mufs.each do |u|
-        expect(filters.find { |f| f.related.email == u[:login] })
-      end
+def test_user_filters_brick(options = {})
+  test_context = options[:test_context]
```

Comment:
info: Refers to 'options' more than self (maybe move it to another class?) (FeatureEnvy) (reek)